From 98716b76d5f78e8ea0670b841ca470dffefb53d9 Mon Sep 17 00:00:00 2001
From: Bruno Haible <bruno@clisp.org>
Date: Tue, 24 Jan 2012 12:22:41 +0100
Subject: [PATCH] stdint: Improve support for Android.

* lib/stdint.in.h: Test __ANDROID__, not __BIONIC__.
Reported by Simon Josefsson <simon@josefsson.org>.
---
 ChangeLog       | 6 ++++++
 lib/stdint.in.h | 8 +++++---
 2 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/srclib/stdint.in.h b/srclib/stdint.in.h
index 591371c..1d0759e 100644
--- a/srclib/stdint.in.h
+++ b/srclib/stdint.in.h
@@ -36,8 +36,10 @@
 
 /* On Android (Bionic libc), <sys/types.h> includes this file before
    having defined 'time_t'.  Therefore in this case avoid including
-   other system header files; just include the system's <stdint.h>.  */
-#if defined __BIONIC__ \
+   other system header files; just include the system's <stdint.h>.
+   Ideally we should test __BIONIC__ here, but it is only defined after
+   <sys/cdefs.h> has been included; hence test __ANDROID__ instead.  */
+#if defined __ANDROID__ \
     && defined _SYS_TYPES_H_ && !defined _SSIZE_T_DEFINED_
 # @INCLUDE_NEXT@ @NEXT_STDINT_H@
 #else
@@ -614,5 +616,5 @@ typedef int _verify_intmax_size[sizeof (intmax_t) == sizeof (uintmax_t)
 #endif
 
 #endif /* _@GUARD_PREFIX@_STDINT_H */
-#endif /* !(defined __BIONIC__ && ...) */
+#endif /* !(defined __ANDROID__ && ...) */
 #endif /* !defined _@GUARD_PREFIX@_STDINT_H && !defined _GL_JUST_INCLUDE_SYSTEM_STDINT_H */
-- 
1.8.4

